packages/clerk-js/src/core/clerk.ts

```diff
@@ -543,84 +541,121 @@ export default class Clerk implements ClerkInterface {
     return await customNavigate(stripOrigin(toURL));
   };
 
-  public redirectToSignIn = async (options?: RedirectOptions): Promise<unknown> => {
+  public buildUrlWithAuth(to: string): string {
+    if (this.#instanceType === 'production' || this.#options.__experimental__devSessionSyncMode === 'cookie') {
+      return to;
+    }
+
+    const toURL = new URL(to, window.location.href);
+
+    if (toURL.origin !== window.location.origin) {
+      const devBrowserJwt = this.#devBrowserHandler?.getDevBrowserJWT();
+      if (!devBrowserJwt) {
+        return clerkMissingDevBrowserJwt();
+      }
+
+      toURL.hash = setSearchParameterInHash({
+        hash: toURL.hash,
+        paramName: DEV_BROWSER_SSO_JWT_PARAMETER,
+        paramValue: devBrowserJwt,
+      });
+    }
+
+    return toURL.href;
+  }
+
+  public buildSignInUrl(options?: RedirectOptions): string {
     const opts: RedirectOptions = {
       ...options,
       redirectUrl: options?.redirectUrl || window.location.href,
     };
     if (!this.#environment || !this.#environment.displayConfig) {
-      return;
+      return '';
     }
     const { signInUrl } = this.#environment.displayConfig;
-    const url = appendAsQueryParams(signInUrl, opts);
-    return this.redirectWithAuth(url);
-  };
+    return this.buildUrlWithAuth(appendAsQueryParams(signInUrl, opts));
+  }
 
-  public redirectToSignUp = async (options?: RedirectOptions): Promise<unknown> => {
+  public buildSignUpUrl(options?: RedirectOptions): string {
     const opts: RedirectOptions = {
       ...options,
       redirectUrl: options?.redirectUrl || window.location.href,
     };
     if (!this.#environment || !this.#environment.displayConfig) {
-      return;
+      return '';
     }
     const { signUpUrl } = this.#environment.displayConfig;
-    const url = appendAsQueryParams(signUpUrl, opts);
-    return this.redirectWithAuth(url);
-  };
+    return this.buildUrlWithAuth(appendAsQueryParams(signUpUrl, opts));
+  }
 
-  public redirectToUserProfile = async (): Promise<unknown> => {
+  public buildUserProfileUrl(): string {
     if (!this.#environment || !this.#environment.displayConfig) {
-      return;
+      return '';
     }
-    return this.redirectWithAuth(this.#environment.displayConfig.userProfileUrl);
-  };
+    return this.buildUrlWithAuth(this.#environment.displayConfig.userProfileUrl);
+  }
 
-  public redirectToHome = async (): Promise<unknown> => {
+  public buildHomeUrl(): string {
     if (!this.#environment || !this.#environment.displayConfig) {
-      return;
+      return '';
     }
-    return this.redirectWithAuth(this.#environment.displayConfig.homeUrl);
-  };
+    return this.buildUrlWithAuth(this.#environment.displayConfig.homeUrl);
+  }
+
+  public buildCreateOrganizationUrl(): string {
+    if (!this.#environment || !this.#environment.displayConfig) {
+      return '';
+    }
```

Comment:
:upside_down_face: Looks like this check is getting repetitive. Perhaps extract a function to check once and reuse?

Something like `buildUrlFromDisplayConfig(property)`.